bundles/org.openhab.binding.paradoxalarm/src/main/java/org/openhab/binding/paradoxalarm/internal/util/ParadoxUtil.java

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.paradoxalarm.internal.util;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ParadoxUtil} Utility class for different calculations / manipulations of data in the model and
+ * communicators.
+ *
+ * @author Konstantin_Polihronov - Initial contribution
+ */
+public class ParadoxUtil {
+
+    private final static Logger logger = LoggerFactory.getLogger(ParadoxUtil.class);
+
+    public static byte calculateChecksum(byte[] payload) {
+        int result = 0;
+        for (byte everyByte : payload) {
+            result += everyByte;
+        }
+
+        return (byte) (result % 256);
+    }
+
+    public static byte getBit(int value, int bitNumber) {
+        return (byte) ((value >> bitNumber) & 1);
+    }
+
+    public static boolean isBitSet(int value, int bitNumber) {
+        return ((value >> bitNumber) & 1) == 1;
+    }
+
+    public static void printPacket(String description, byte[] array) {
+        logger.trace("Packet payload size: {}", array[1]);

```

Comment:
Should this method be completely wrapped in a check whether trace logging is enabled?